templates/common/render/action_include.go

```diff
@@ -138,17 +137,50 @@ func isGlob(matchedPaths []model.String, originalPath, matchedPath string) bool
 }
 
 func includePath(ctx context.Context, inc *spec.IncludePath, sp *stepParams) error {
-	// By default, we copy from the template directory. We also support
-	// grabbing files from the destination directory, so we can modify files
-	// that already exist in the destination.
-	fromDir := sp.templateDir
+	// By default, we copy from the template directory.
+	fromDirs := []string{sp.templateDir}
 	if inc.From.Val == "destination" {
-		fromDir = sp.rp.DestDir
+		// We also support including files from the destination directory, so we
+		// can modify files that already exist in the destination.
+		fromDirs = []string{sp.rp.DestDir}
+		// TODO(upgrade): add this field in an upcoming PR
+		// if sp.rp.IncludeFromDestExtraDir != "" {
+		// For complicated reasons related to upgrading, we sometimes add
+		// another include source directory that contains files after having
+		// had their include-from-destination patch reversed as part of the
+		// upgrade process.
+		//	fromDirs = append(fromDirs, sp.rp.IncludeFromDestExtraDir)
+		// }
+	}
+
+	anyMatches := false
```

Comment:
Feels like I've seen this change in the other PR you have.

Reply:
Oops, I didn't set the base branch on this PR properly, sorry. This should resolve itself when I merge the parent PR.